`mozjs` fails to build from source on GitHub hosted runners

last good: https://github.com/servo/servo/actions/runs/12255916416

first busted: https://github.com/servo/servo/actions/runs/12261266973/job/34207914801

between them https://github.com/servo/servo/pull/34562 opened, so maybe it busted some caches or smth, can somebody try to clear github CI caches to see if this fixes the issue.

> I also saw the issue on non-try branches for the regular Linux build job.

Can't that be explained by the fact that all workflows can fallback to the caches of the default (main) branch? The doc warns against saving the cache, presumably because PRs can then write corrupt cache entries to their base branch, which for our PRs is just the `main` branch usually.

But still, I am not sure how that explains the error we are seeing, which seems to be related to `make` issuing a wrong compilation command.

```
2024-12-11T08:42:30.8525685Z   make[4]: *** No rule to make target '/home/runner/.cargo/git/checkouts/mozjs-76be79f630786c2e/daad60b/mozjs-sys/mozjs/build/pure_virtual/pure_virtual.c', needed by 'pure_virtual.o'.  Stop.
2024-12-11T08:42:30.8526903Z   make[3]: *** [/home/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/18ab014/mozjs-sys/mozjs/config/recurse.mk:72: build/pure_virtual/target] Error 2
```

The path to `mozjs-sys` source code is different in both lines. The first is under `.../mozjs-76be79f630786c2e/daad60b/..` path while the actual mozjs checkout that the rest of the build uses is under `../mozjs-fa11ffc7d4f1cc2d/18ab014/`. 

The *No rule to make target* error is the only one that references the` ../mozjs-76be79f630786c2e/daad60b/..` path in the whole [build log](https://productionresultssa7.blob.core.windows.net/actions-results/87260054-a42c-4dd6-9b9d-1059e9c16bf0/workflow-job-run-644610c7-6215-5239-0278-ebe8d087ccbb/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-12-15T16%3A47%3A44Z&sig=jwb8SXCkp4bWv36WhuWjLHKgUEDyH1dQESgZvApwCfQ%3D&ske=2024-12-16T01%3A47%3A30Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-12-15T13%3A47%3A30Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-11-04&sp=r&spr=https&sr=b&st=2024-12-15T16%3A37%3A39Z&sv=2024-11-04).

--- a/.github/workflows/android.yml
+++ b/.github/workflows/android.yml
@@ -24,11 +24,6 @@ on:
 env:
   RUST_BACKTRACE: 1
   SHELL: /bin/bash
-  SCCACHE_GHA_ENABLED: "true"
-  RUSTC_WRAPPER: "sccache"
-  CCACHE: "sccache"
-  # Increment this to purge the cache (https://github.com/mozilla/sccache/blob/main/docs/GHA.md)
-  SCCACHE_GHA_VERSION: 3
   CARGO_INCREMENTAL: 0
 
 jobs:
@@ -57,8 +52,6 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 2
-      - name: Run sccache-cache
-        uses: mozilla-actions/sccache-action@v0.0.7
       - name: Install crown
         run: cargo install --path support/crown
       - name: Setup Python

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -6,11 +6,6 @@ on:
 env:
   RUST_BACKTRACE: 1
   SHELL: /bin/bash
-  SCCACHE_GHA_ENABLED: "true"
-  # Increment this to purge the cache (https://github.com/mozilla/sccache/blob/main/docs/GHA.md)
-  SCCACHE_GHA_VERSION: 3
-  RUSTC_WRAPPER: "sccache"
-  CCACHE: "sccache"
   CARGO_INCREMENTAL: 0
 
 jobs:
@@ -29,8 +24,6 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 2 # This is necessary for `test-tidy`.
-      - name: Run sccache-cache
-        uses: mozilla-actions/sccache-action@v0.0.6
       - name: Set LIBCLANG_PATH env # needed for bindgen in mozangle
         run: echo "LIBCLANG_PATH=/usr/lib/llvm-14/lib" >> $GITHUB_ENV
       - name: Setup Python

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -75,8 +75,6 @@ on:
 env:
   RUST_BACKTRACE: 1
   SHELL: /bin/bash
-  # Increment this to purge the cache (https://github.com/mozilla/sccache/blob/main/docs/GHA.md)
-  SCCACHE_GHA_VERSION: 3
 
 jobs:
   # Runs the underlying job (“workload”) on a self-hosted runner if available,
@@ -106,13 +104,6 @@ jobs:
     name: Linux Build [${{ needs.runner-select.outputs.unique-id }}]
     runs-on: ${{ needs.runner-select.outputs.selected-runner-label }}
     steps:
-      - if: ${{ runner.environment != 'self-hosted' }}
-        run: |
-          echo SCCACHE_GHA_ENABLED=true >> $GITHUB_ENV
-          echo RUSTC_WRAPPER=sccache >> $GITHUB_ENV
-          echo CCACHE=sccache >> $GITHUB_ENV
-          echo CARGO_INCREMENTAL=0 >> $GITHUB_ENV
-
       - uses: actions/checkout@v4
         if: ${{ runner.environment != 'self-hosted' && github.event_name != 'pull_request_target' }}
       # This is necessary to checkout the pull request if this run was triggered via a
@@ -134,10 +125,6 @@ jobs:
           git switch --detach
           git reset --hard FETCH_HEAD
 
-      # Install missing tools in a GitHub-hosted runner.
-      - name: Run sccache-cache
-        if: ${{ runner.environment != 'self-hosted' }}
-        uses: mozilla-actions/sccache-action@v0.0.6
       - name: Set LIBCLANG_PATH env # needed for bindgen in mozangle
         if: ${{ runner.environment != 'self-hosted' && !inputs.upload }} # not needed on ubuntu 20.04 used for nightly
         run: echo "LIBCLANG_PATH=/usr/lib/llvm-14/lib" >> $GITHUB_ENV

--- a/.github/workflows/mac.yml
+++ b/.github/workflows/mac.yml
@@ -52,11 +52,6 @@ on:
 env:
   RUST_BACKTRACE: 1
   SHELL: /bin/bash
-  # Increment this to purge the cache (https://github.com/mozilla/sccache/blob/main/docs/GHA.md)
-  SCCACHE_GHA_VERSION: 4
-  SCCACHE_GHA_ENABLED: "true"
-  RUSTC_WRAPPER: "sccache"
-  CCACHE: "sccache"
   CARGO_INCREMENTAL: 0
 
 jobs:
@@ -81,8 +76,6 @@ jobs:
           ref: ${{ github.event.pull_request.head.sha }}
       - name: Setup Python
         uses: ./.github/actions/setup-python
-      - name: Run sccache-cache
-        uses: mozilla-actions/sccache-action@v0.0.6
       - name: Install crown
         run: cargo install --path support/crown
       - name: Bootstrap

--- a/.github/workflows/ohos.yml
+++ b/.github/workflows/ohos.yml
@@ -25,11 +25,6 @@ on:
 env:
   RUST_BACKTRACE: 1
   SHELL: /bin/bash
-  # Increment this to purge the cache (https://github.com/mozilla/sccache/blob/main/docs/GHA.md)
-  SCCACHE_GHA_VERSION: 3
-  SCCACHE_GHA_ENABLED: "true"
-  RUSTC_WRAPPER: "sccache"
-  CCACHE: "sccache"
   CARGO_INCREMENTAL: 0
 
 jobs:
@@ -51,8 +46,6 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 2
-      - name: Run sccache-cache
-        uses: mozilla-actions/sccache-action@v0.0.6
       - name: Install crown
         run: cargo install --path support/crown
       - name: Setup Python

